contracts/Zauction.sol

```diff
@@ -121,6 +132,77 @@ contract ZAuction is Initializable, OwnableUpgradeable {
     );
   }
 
+  function setBuyPrice(uint256 amount, uint256 tokenId) external {
+    address owner = registrar.ownerOf(tokenId);
+    require(msg.sender == owner, "zAuction: only owner can set price");
+    buyPrice[tokenId] = amount;
+  }
+
+  /// recovers buyer's signature based on seller's proposed data and, if bid data hash matches the message hash, transfers nft and payment
+  /// @param signature type encoded message signed by the buyer
+  /// @param auctionId unique per address auction identifier chosen by seller
+  /// @param amount token amount of sale
+  /// @param tokenId token id we are transferring
+  /// @param startBlock block number at which acceptBid starts working
+  /// @param expireBlock block number at which acceptBid stops working
+  function buyNow(
+    bytes memory signature,
```

Comment:
Why does this function need a signature / signed message?
Who is calling this function? The buyer or the seller? If it's the buyer, why do they need to sign anything? They are the ones making the purchase as well as the transaction. Signing a message is only needed if you want to 'authorize' someone to do something on your behalf. (seller signs message, buy does the transaction using seller signed message)